conjur/init/init_logic.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+"""
+InitLogic module
+
+This module is the business logic for writing configuration information
+to the user's machine as well as fetching certificates from Conjur
+"""
+
+# Builtins
+import logging
+import os.path
+
+# Third party
+import yaml
+
+import conjur.constants
+from conjur.endpoints import ConjurEndpoint
+from conjur.http import invoke_endpoint, HttpVerb
+
+DEFAULT_PORT = 443
+
+# pylint: disable=raise-missing-from
+class InitLogic:
+    """
+    InitLogic
+
+    This class holds the business logic for populating the
+    conjurrc configuration details needed to connect to Conjur
+    """
+    def __init__(self, ssl_service):
+        self.ssl_service = ssl_service
+
+    def get_certificate(self, hostname, port):
+        """
+        Method for connecting to Conjur to fetch the certificate chain
+        """
+        if port is None:
+            port = DEFAULT_PORT
+        try:
+            fingerprint, readable_certificate = self.ssl_service.get_certificate(hostname, port)
+            logging.debug("Successfully fetched certificate")
+        except Exception as error:
+            raise Exception(f"Unable to retrieve certificate from {hostname}:{port}. " \
+                            f"Reason: {str(error)}") from error
+
+        return fingerprint, readable_certificate
+
+    @staticmethod
+    def fetch_account_from_server(conjurrc_data):
+        """
+        Fetches the account from the DAP server by making a request to the /info endpoint.
+        This endpoint only exists in the DAP server
+        """
+        params = {
+            'url': conjurrc_data.appliance_url
+        }
+        # If the user provides us with the certificate path, we will use it
+        # to make a request to /info
+        if conjurrc_data.cert_file is None and conjurrc_data.appliance_url.startswith("https"):
+            certificate_path = os.path.join(os.path.dirname(conjur.constants.DEFAULT_CONFIG_FILE),
+                                          "conjur-server.pem")
+        else:
+            certificate_path = conjurrc_data.cert_file
+
+        logging.debug("Attempting to fetch the account from the Conjur server")
+        response = invoke_endpoint(HttpVerb.GET,
+                                   ConjurEndpoint.INFO,
+                                   params,
+                                   ssl_verify=certificate_path).json()
```

Comment:
for getting account info, you should always use `conjurrc_data.cert_file`.
Comment why it is correct to do it here (depend on `write certificate` done in previous stage).
```suggestion
                                   ssl_verify=conjurrc_data.cert_file).json()
```